src/lib.rs

```diff
@@ -2,6 +2,7 @@ use crate::error::Error;
 use crate::source::Source;
 
 pub mod error;
+pub(crate) mod hash;
```

Comment:
```suggestion
pub mod hash;
```

there's no need to keep this internal.